openwisp_network_topology/api/views.py

```diff
@@ -16,7 +23,32 @@
 Topology = swapper.load_model('topology', 'Topology')
 
 
-class NetworkCollectionView(generics.ListAPIView):
+class CustomDjangoModelPermissions(DjangoModelPermissions):
+    """
+    Add GET method permission check to DjangoModelPermissions.
+    """
+
+    def __init__(self):
+        self.perms_map = deepcopy(self.perms_map)
+        self.perms_map['GET'] = [
+            f'{Topology._meta.app_label.lower()}.view_{Topology._meta.object_name.lower()}'
+        ]
```

Comment:
why do we need this? Maybe I forgot something.

Reply:
From [django_rest_framework documentation](https://www.django-rest-framework.org/api-guide/permissions/#djangomodelpermissions), `DjangoModelPermissions` doesn't work for the `GET` method. so we have to created a custom permission which takes care of `view` permission. this ensures that in API endpoints where GET method is used, the user sees only objects which he has the `view permissions` on

Reply:
> From [django_rest_framework documentation](https://www.django-rest-framework.org/api-guide/permissions/#djangomodelpermissions), `DjangoModelPermissions` doesn't work for the `GET` method. so we have to created a custom permission which takes care of `view` permission. this ensures that in API endpoints where GET method is used, the user sees only objects which he has the `view permissions` on

Thanks for the clarification :+1: